GrafeasGroup/tor_archivist
# Dev pm

Relevant issue: https://github.com/GrafeasGroup/tor_archivist/issues/14

## Description:

Acceptance Criteria:
- Bot retains same rules and settings (e.g., grace period) determining when to archive a post 
- Bot uses Blossom's API to receive work instead of Reddit's API 
- Bot only attempts to archive posts that are not yet archived 

Updated flow:
u/ToR performs the validation, so it will post the human-generated transcription to Blossom. The full process should look like this:

- Ask Blossom for submissions that are complete and have not yet been archived 
- Request transcription link for submission from Blossom 
- Remove submission on r/ToR
- Post transcription URL as a new submission on r/ToR_Archives 
- Inform Blossom that the submission has been archived 

For posts that have expired (passed the 18 hour threshold):

- request list of submission objects to remove 
- for each submission object:
  - remove from r/ToR 
  - inform Blossom that the post has been archived 

## Screenshots:

## Testing Instructions:
- start local instance of blossom API
- if there are unarchived submissions, they should show up at the `/submissions/unarchived/` endpoint.
- if there are expired submissions, they should show up at the `/submissions/expired/` endpoint.
- run tor-archivist

for each submission at the unarchived endpoint, tor_archivist should now:
- mark the submission as archived
- get the transcription for that submission and post it to reddit

for each submission at the expired endpoint, tor_archivist should now:
- mark the submission as archived

Note: did not test removing original post from reddit due to lack of mod access. In production be sure to uncomment the lines which remove the original post (line 62 and 93). Also, change config.archive to tor_archive instead of tor_testing_ground (replace line 111 with 114)

## Checklist:
